snuba/migrations/snuba_migrations/generic_metrics/0001_sets_aggregate_table.py

```diff
@@ -0,0 +1,143 @@
+                    "_indexed_tags_hash",
+                    Array(
+                        UInt(64),
+                        Modifiers(
+                            materialized=hash_map_int_column_definition("indexed_tags")
+                        ),
+                    ),
+                ),
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+                column=Column(
+                    "_raw_tags_hash",
+                    Array(
+                        UInt(64),
+                        Modifiers(
+                            materialized=hash_map_int_key_str_value_column_definition(
+                                "raw_tags"
+                            )
+                        ),
+                    ),
+                ),
+            ),
+            operations.AddIndex(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+                index_name="bf_indexed_tags_hash",
+                index_expression="_indexed_tags_hash",
+                index_type="bloom_filter()",
+                granularity=1,
+            ),
+            operations.AddIndex(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+                index_name="bf_indexed_tags_key_hash",
+                index_expression="indexed_tags.key",
+                index_type="bloom_filter()",
+                granularity=1,
+            ),
+            operations.AddIndex(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+                index_name="bf_raw_tags_hash",
+                index_expression="_raw_tags_hash",
+                index_type="bloom_filter()",
+                granularity=1,
+            ),
+            operations.AddIndex(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+                index_name="bf_raw_tags_key_hash",
+                index_expression="raw_tags.key",
+                index_type="bloom_filter()",
+                granularity=1,
+            ),
+        ]
+
+    def backwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.DropTable(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+            )
+        ]
+
+    def forwards_dist(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name="generic_metric_sets_aggregated_dist",
+                engine=table_engines.Distributed(
+                    local_table_name=self.local_table_name,
+                    sharding_key="timeseries_id",
```

Comment:
You do not need to add this sharding key to the aggregated distributed table. 
That goes on the raw table as materialization happens on local nodes. Not distributed ones.